server/src/main/java/com/cloud/api/ApiResponseHelper.java

```diff
@@ -992,6 +992,9 @@ private void addSystemVmInfoToIpResponse(NicVO nic, IPAddressResponse ipResponse
         if (!isAdmin) {
             return;
         }
+        if (Nic.ReservationStrategy.PlaceHolder.equals(nic.getReservationStrategy())) {
```

Comment:
@weizhouapache , these are 4 more potential NPEs, did you check those?

Reply:
Did you mean this code snippet or some where else?

Reply:
in the condition here. I'll check them tomorrow.

Reply:
it is defined as
```
    public enum ReservationStrategy {
        PlaceHolder, Create, Start, Managed;
    }
```

Reply:
it is defined as below
```
    public enum ReservationStrategy {
        PlaceHolder, Create, Start, Managed;
    }
```

Reply:
sorry for the late reply @weizhouapache yes and nic as well would have thrown an NPE before. I think it is not intuitive code but you fix is good and an improvement on it.

Reply:
Is it ok now?